### DISCUSSION TITLE
Convergence factors and issues with scaling of devices

### DISCUSSION THREAD

**Discussion by chuanhong98** [facebookresearch/fmmax#123] [Q&A]
Hello Martin / developers,

I am using fmmax solely to simulate µLEDs (based on the paper you have released and the example given from git). 

After looking into a few factors that affect convergence, namely number of Fourier orders, FWHM of Gaussian and planes where the Poynting flux is summed to obtain the power emitted by dipole I am simulating µLEDs with different sizes and outcoupling structures.

I have come across some situations where the extraction efficiency is more than 1, but I have just accounted this as issue due to convergence that subsided when I increased the Fourier orders _N_ until the extraction efficiency has converged and seems reasonable (lesser than 1). Have you encountered such issues before? Do you also perhaps know how Fourier orders _N_ would scale according to device sizes?

Other than that I have a question regarding a structure as below,
![image](https://github.com/facebookresearch/fmmax/assets/168533583/9c0b6d29-9ed1-46fd-8da6-e86adffcc710)
where the dipole is placed in an active region with thick passivation. Will the calculation of the emitted power (summing of poynting flux) be affected due to large changes in permittivities? As I assume that the plane taken is across the whole pitch (across different permittivities).

I thank you for your time and look forward to hearing from you again. :)

**Comment by mfschubert** [comment]
Hi @chuanhong98 , I recall having seen this but also managed to get rid of the problem. Here are some suggestions:
- increase the number of terms in the Fourier expansion. This should scale with the area of the unit cell, i.e. 2x area -> 2x terms
- try a different formulation. Nowadays, I am typically using JONES_DIRECT_FOURIER
- add a small amount of loss to dielectric materials. This helps ensure that eigenmodes which carry power forwards are correctly categorized as "forward" modes. Sometimes, spurious imaginary components to the eigenvalues result in misclassification, and I believe this may be related to your >1 extraction efficiency.
- try 64 bit mode. This also helps minimize the spurious imaginary components to eigenvalues.

I'm not sure I understand your question about dipole power, but I would expect a dipole in free space vs. your structure to be substantially different because there is effectively a 2D cavity here.

**Reply by chuanhong98** [reply]
hi @mfschubert, thank you for your suggestions and quick replies as always.

- I have been using JONES_DIRECT vector formulation up until now. Will try the JONES_DIRECT_FOURIER now
- adding a small amount of loss to dielectric materials also sounded like a good idea

I will try them out and get back to you on the results. Regarding the dipole power I realised I have not formulated my questions properly.

I am just curious if the total_emitted_power calculated across the monitors will be affected by the non-uniform permittivities across the layers as show in the figure below. I thought about this question as the total_extracted_power is calculated in a medium (ambient) where the permittivities across it is uniform

![image](https://github.com/facebookresearch/fmmax/assets/168533583/6006c2e0-86b7-4690-ac14-3c84c7f93fe4)


Thank you once again! :)

**Reply by mfschubert** [reply]
@chuanhong98, I'd say the power is affected to the extent that the emission of the dipole itself is modified by the surrounding structure. In general, the structure should have some effect, e.g. it could create a cavity that enhances or suppresses the dipole emission. So, yes---the total emitted power would likely differ from the uniform permittivity case.
